helix-core/src/movement.rs

```diff
@@ -64,7 +64,6 @@ pub fn move_vertically_visual(
     if !text_fmt.soft_wrap {
         return move_vertically(slice, range, dir, count, behaviour, text_fmt, annotations);
     }
-    annotations.clear_line_annotations();
```

Comment:
I want the cursor and page to jump by the same amount of vertical distance on each scroll. 
My function calls this method to move the cursor, if this line wasnt removed it would be moving the cursor a distance of `count` without respect for complete lines of virtual text (inline diagnostics) (eg which are present in #6417 ), meaning it would be jumping a greater visual distance that the view is. (As the view calls `char_idx_at_visual_offset` which respects whole lines of virtual text).

I've instead called this line in `move_impl` so that functions like `move_visual_line_down` will not land the cursor on complete lines of visual text/inlnie diagnostics.

Reply:
You probably need to handle that differently. There is a fast oath for the case that softwrap is disabled and it doesn't respect multilinear virtual text anyway (it just uses line numbers).

Reply:
a fast oath?

Reply:
https://github.com/helix-editor/helix/blob/01a1e5ec2a8c5b81da1360746dad0a96cd88ae8e/helix-core/src/movement.rs#L64

Reply:
like this [e490572](https://github.com/helix-editor/helix/pull/8015/commits/e4905729c338a2260e6981f1d8fac022897b4191)? I am getting inconsistent jumping when the cursor starts on the non-virtual line directly above a multiline diagnostic, the scroll up is inconsistent depending on which word in the line the cursor starts on. Not sure what is causing this

Reply:
yeah, that should work. Honestly, I think for now you can just use the existing `move_vertically_visual` function without modifying it. I will look into this when I get back to the inline diagnostic PR. That PR still has many bugs so its not clear if that is actually a problem with you code.

Reply:
ok [3cc3e2a](https://github.com/helix-editor/helix/pull/8015/commits/3cc3e2a489a7f880fa3ffae7fade441289f95778), this small refactor does also allow me to easily solve the 'constant distance between view and cursor if cursor lands on whole line of virtual text' that i was talking about in the description. But this can also wait until the inline diagnostics pr gets fixed/merged too.